plugins/bnr-dp/fu-bnr-dp-common.h

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright 2024-2025 B&R Industrial Automation GmbH
+ *
+ * SPDX-License-Identifier: LGPL-2.1-or-later
+ */
+
+#pragma once
+
+#include <glib.h>
+
+#include "fu-bnr-dp-struct.h"
+
+gchar *
+fu_bnr_dp_version_to_string(guint64 version);
+
+gboolean
+fu_bnr_dp_version_from_header(const FuStructBnrDpPayloadHeader *header,
+			      guint64 *version,
+			      GError **error);
```

Comment:
I think adding  `G_GNUC_NON_NULL(1, 2)` before the `;` will give you a little more compile-time testing -- same for all the other stuff in this header.